src/server/transaction.cc

```diff
@@ -1445,12 +1445,14 @@ OpResult<KeyIndex> DetermineKeys(const CommandId* cid, CmdArgList args) {
     if (!absl::SimpleAtoi(num, &num_custom_keys) || num_custom_keys < 0)
       return OpStatus::INVALID_INT;
 
-    // TODO Fix this for Z family functions.
-    // Examples that crash: ZUNION 0 myset
     if (name == "ZDIFF" && num_custom_keys == 0) {
       return OpStatus::INVALID_INT;
     }
 
+    if (name == "ZUNION" && num_custom_keys == 0) {
+      return OpStatus::INVALID_INT;
+    }
+
```

Comment:
you can squash those two in one condition:

```
if ((name == "ZUNION" || name == "ZDIFF") && num_custom_keys == 0)
```

Reply:
Not squashing since we are returning `OpStatus::SYNTAX_ERR`